dev/release/VERIFY.md

```diff
@@ -74,3 +74,36 @@ You need to set `PKG_CONFIG_PATH` to find libffi on macOS:
 ```console
 % export PKG_CONFIG_PATH=$(brew --prefix libffi)/lib/pkgconfig:$PKG_CONFIG_PATH
 ```
+
+### C++, C#, C GLib, Go, JavaScript, Python, Ruby
+
+Example scripts to install the dependencies to run the verification 
+script for verifying the source on Ubuntu 20.04, Rocky Linux 8 and 
+AlmaLinux 8 are in this folder and named `setup-ubuntu.sh` and 
+`setup-rhel-derivatives.sh`. These can be adapted to different 
+situations. Go and JavaScript are installed by the verification 
+
+script in the testing environment. Verifying the binaries additionally
+requires installation of Docker. Verifying the wheels requires 
+installation of packages listed in `../../python/requirements-test.txt`, 
+and additionally requires running
+
+```console
+git submodule init
+git submodule update
+```
```

Comment:
This instruction isn't needed. Because:

1. The verification script verifies in `apache-arrow-X.Y.Z/` that is created by extraction of `apache-arrow-X.Y.Z.tar.gz` not the checkout-ed directory. (No `.git/` in `apache-arrow-X.Y.Z/'.)
2. The verification script clones test data automatically: https://github.com/apache/arrow/blob/master/dev/release/verify-release-candidate.sh#L703

Reply:
Without this, the verification failed on a fresh install of RockyLinux 8

Reply:
Could you show instruction how to reproduce it?

Reply:
Commands are
```console
dnf -y install \
  'dnf-command(config-manager)'
dnf config-manager --set-enabled powertools
dnf -y module disable ruby
dnf -y module enable ruby:2.7
dnf -y groupinstall "Development Tools"
dnf -y install \
  cmake \
  epel-release \
  gdbm-devel \
  git \
  gobject-introspection-devel \
  java-1.8.0-openjdk-devel \
  libcurl-devel \
  llvm-devel \
  llvm-toolset \
  maven \
  ncurses-devel \
  ninja-build \
  openssl-devel \
  python3-pip \
  python3-devel \
  ruby-devel \
  wget 
alternatives --set python /usr/bin/python3

git clone https://github.com/apache/arrow
cd arrow/dev/release
./verify-release-candidate.sh wheels 6.0.1 1
```

The following test does not pass without adding the submodule
```console
================================================ FAILURES ================================================
____________________________________________ test_tls_do_get _____________________________________________

path = 'root-ca.pem'

    def read_flight_resource(path):
        """Get the contents of a test resource file."""
        root = resource_root()
        if not root:
            return None
        try:
>           with (root / path).open("rb") as f:

../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/site-packages/pyarrow/tests/test_flight.py:77: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = PosixPath('/root/arrow/testing/data/flight/root-ca.pem'), mode = 'rb', buffering = -1
encoding = None, errors = None, newline = None

    def open(self, mode='r', buffering=-1, encoding=None,
             errors=None, newline=None):
        """
        Open the file pointed by this path and return a file object, as
        the built-in open() function does.
        """
        if self._closed:
            self._raise_closed()
        return io.open(str(self), mode, buffering, encoding, errors, newline,
>                      opener=self._opener)

../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/pathlib.py:1183: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = PosixPath('/root/arrow/testing/data/flight/root-ca.pem')
name = '/root/arrow/testing/data/flight/root-ca.pem', flags = 524288, mode = 438

    def _opener(self, name, flags, mode=0o666):
        # A stub for the opener argument to built-in open()
>       return self._accessor.open(self, flags, mode)

../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/pathlib.py:1037: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

pathobj = PosixPath('/root/arrow/testing/data/flight/root-ca.pem'), args = (524288, 438)

    @functools.wraps(strfunc)
    def wrapped(pathobj, *args):
>       return strfunc(str(pathobj), *args)
E       FileNotFoundError: [Errno 2] No such file or directory: '/root/arrow/testing/data/flight/root-ca.pem'

../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/pathlib.py:387: FileNotFoundError

During handling of the above exception, another exception occurred:

    @pytest.mark.requires_testing_data
    def test_tls_do_get():
        """Try a simple do_get call over TLS."""
        table = simple_ints_table()
>       certs = example_tls_certs()

../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/site-packages/pyarrow/tests/test_flight.py:1345: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/site-packages/pyarrow/tests/test_flight.py:89: in example_tls_certs
    "root_cert": read_flight_resource("root-ca.pem"),
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

path = 'root-ca.pem'

    def read_flight_resource(path):
        """Get the contents of a test resource file."""
        root = resource_root()
        if not root:
            return None
        try:
            with (root / path).open("rb") as f:
                return f.read()
        except FileNotFoundError:
            raise RuntimeError(
                "Test resource {} not found; did you initialize the "
                "test resource submodule?\n{}".format(root / path,
>                                                     traceback.format_exc()))
E           RuntimeError: Test resource /root/arrow/testing/data/flight/root-ca.pem not found; did you initialize the test resource submodule?
E           Traceback (most recent call last):
E             File "/tmp/arrow-6.0.1.edIR4/test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/site-packages/pyarrow/tests/test_flight.py", line 77, in read_flight_resource
E               with (root / path).open("rb") as f:
E             File "/tmp/arrow-6.0.1.edIR4/test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/pathlib.py", line 1183, in open
E               opener=self._opener)
E             File "/tmp/arrow-6.0.1.edIR4/test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/pathlib.py", line 1037, in _opener
E               return self._accessor.open(self, flags, mode)
E             File "/tmp/arrow-6.0.1.edIR4/test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/pathlib.py", line 387, in wrapped
E               return strfunc(str(pathobj), *args)
E           FileNotFoundError: [Errno 2] No such file or directory: '/root/arrow/testing/data/flight/root-ca.pem'

../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/site-packages/pyarrow/tests/test_flight.py:83: RuntimeError
__________________________________ test_tls_disable_server_verification __________________________________

path = 'root-ca.pem'

    def read_flight_resource(path):
        """Get the contents of a test resource file."""
        root = resource_root()
        if not root:
            return None
        try:
>           with (root / path).open("rb") as f:

../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/site-packages/pyarrow/tests/test_flight.py:77: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = PosixPath('/root/arrow/testing/data/flight/root-ca.pem'), mode = 'rb', buffering = -1
encoding = None, errors = None, newline = None

    def open(self, mode='r', buffering=-1, encoding=None,
             errors=None, newline=None):
        """
        Open the file pointed by this path and return a file object, as
        the built-in open() function does.
        """
        if self._closed:
            self._raise_closed()
        return io.open(str(self), mode, buffering, encoding, errors, newline,
>                      opener=self._opener)

../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/pathlib.py:1183: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = PosixPath('/root/arrow/testing/data/flight/root-ca.pem')
name = '/root/arrow/testing/data/flight/root-ca.pem', flags = 524288, mode = 438

    def _opener(self, name, flags, mode=0o666):
        # A stub for the opener argument to built-in open()
>       return self._accessor.open(self, flags, mode)

../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/pathlib.py:1037: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

pathobj = PosixPath('/root/arrow/testing/data/flight/root-ca.pem'), args = (524288, 438)

    @functools.wraps(strfunc)
    def wrapped(pathobj, *args):
>       return strfunc(str(pathobj), *args)
E       FileNotFoundError: [Errno 2] No such file or directory: '/root/arrow/testing/data/flight/root-ca.pem'

../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/pathlib.py:387: FileNotFoundError

During handling of the above exception, another exception occurred:

    @pytest.mark.requires_testing_data
    def test_tls_disable_server_verification():
        """Try a simple do_get call over TLS with server verification disabled."""
        table = simple_ints_table()
>       certs = example_tls_certs()

../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/site-packages/pyarrow/tests/test_flight.py:1358: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/site-packages/pyarrow/tests/test_flight.py:89: in example_tls_certs
    "root_cert": read_flight_resource("root-ca.pem"),
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

path = 'root-ca.pem'

    def read_flight_resource(path):
        """Get the contents of a test resource file."""
        root = resource_root()
        if not root:
            return None
        try:
            with (root / path).open("rb") as f:
                return f.read()
        except FileNotFoundError:
            raise RuntimeError(
                "Test resource {} not found; did you initialize the "
                "test resource submodule?\n{}".format(root / path,
>                                                     traceback.format_exc()))
E           RuntimeError: Test resource /root/arrow/testing/data/flight/root-ca.pem not found; did you initialize the test resource submodule?
E           Traceback (most recent call last):
E             File "/tmp/arrow-6.0.1.edIR4/test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/site-packages/pyarrow/tests/test_flight.py", line 77, in read_flight_resource
E               with (root / path).open("rb") as f:
E             File "/tmp/arrow-6.0.1.edIR4/test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/pathlib.py", line 1183, in open
E               opener=self._opener)
E             File "/tmp/arrow-6.0.1.edIR4/test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/pathlib.py", line 1037, in _opener
E               return self._accessor.open(self, flags, mode)
E             File "/tmp/arrow-6.0.1.edIR4/test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/pathlib.py", line 387, in wrapped
E               return strfunc(str(pathobj), *args)
E           FileNotFoundError: [Errno 2] No such file or directory: '/root/arrow/testing/data/flight/root-ca.pem'

../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/site-packages/pyarrow/tests/test_flight.py:83: RuntimeError
_______________________________________ test_tls_override_hostname _______________________________________

path = 'root-ca.pem'

    def read_flight_resource(path):
        """Get the contents of a test resource file."""
        root = resource_root()
        if not root:
            return None
        try:
>           with (root / path).open("rb") as f:

../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/site-packages/pyarrow/tests/test_flight.py:77: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = PosixPath('/root/arrow/testing/data/flight/root-ca.pem'), mode = 'rb', buffering = -1
encoding = None, errors = None, newline = None

    def open(self, mode='r', buffering=-1, encoding=None,
             errors=None, newline=None):
        """
        Open the file pointed by this path and return a file object, as
        the built-in open() function does.
        """
        if self._closed:
            self._raise_closed()
        return io.open(str(self), mode, buffering, encoding, errors, newline,
>                      opener=self._opener)

../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/pathlib.py:1183: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = PosixPath('/root/arrow/testing/data/flight/root-ca.pem')
name = '/root/arrow/testing/data/flight/root-ca.pem', flags = 524288, mode = 438

    def _opener(self, name, flags, mode=0o666):
        # A stub for the opener argument to built-in open()
>       return self._accessor.open(self, flags, mode)

../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/pathlib.py:1037: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

pathobj = PosixPath('/root/arrow/testing/data/flight/root-ca.pem'), args = (524288, 438)

    @functools.wraps(strfunc)
    def wrapped(pathobj, *args):
>       return strfunc(str(pathobj), *args)
E       FileNotFoundError: [Errno 2] No such file or directory: '/root/arrow/testing/data/flight/root-ca.pem'

../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/pathlib.py:387: FileNotFoundError

During handling of the above exception, another exception occurred:

    @pytest.mark.requires_testing_data
    def test_tls_override_hostname():
        """Check that incorrectly overriding the hostname fails."""
>       certs = example_tls_certs()

../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/site-packages/pyarrow/tests/test_flight.py:1373: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/site-packages/pyarrow/tests/test_flight.py:89: in example_tls_certs
    "root_cert": read_flight_resource("root-ca.pem"),
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

path = 'root-ca.pem'

    def read_flight_resource(path):
        """Get the contents of a test resource file."""
        root = resource_root()
        if not root:
            return None
        try:
            with (root / path).open("rb") as f:
                return f.read()
        except FileNotFoundError:
            raise RuntimeError(
                "Test resource {} not found; did you initialize the "
                "test resource submodule?\n{}".format(root / path,
>                                                     traceback.format_exc()))
E           RuntimeError: Test resource /root/arrow/testing/data/flight/root-ca.pem not found; did you initialize the test resource submodule?
E           Traceback (most recent call last):
E             File "/tmp/arrow-6.0.1.edIR4/test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/site-packages/pyarrow/tests/test_flight.py", line 77, in read_flight_resource
E               with (root / path).open("rb") as f:
E             File "/tmp/arrow-6.0.1.edIR4/test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/pathlib.py", line 1183, in open
E               opener=self._opener)
E             File "/tmp/arrow-6.0.1.edIR4/test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/pathlib.py", line 1037, in _opener
E               return self._accessor.open(self, flags, mode)
E             File "/tmp/arrow-6.0.1.edIR4/test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/pathlib.py", line 387, in wrapped
E               return strfunc(str(pathobj), *args)
E           FileNotFoundError: [Errno 2] No such file or directory: '/root/arrow/testing/data/flight/root-ca.pem'

../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/site-packages/pyarrow/tests/test_flight.py:83: RuntimeError
_______________________________________________ test_mtls ________________________________________________

path = 'root-ca.pem'

    def read_flight_resource(path):
        """Get the contents of a test resource file."""
        root = resource_root()
        if not root:
            return None
        try:
>           with (root / path).open("rb") as f:

../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/site-packages/pyarrow/tests/test_flight.py:77: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = PosixPath('/root/arrow/testing/data/flight/root-ca.pem'), mode = 'rb', buffering = -1
encoding = None, errors = None, newline = None

    def open(self, mode='r', buffering=-1, encoding=None,
             errors=None, newline=None):
        """
        Open the file pointed by this path and return a file object, as
        the built-in open() function does.
        """
        if self._closed:
            self._raise_closed()
        return io.open(str(self), mode, buffering, encoding, errors, newline,
>                      opener=self._opener)

../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/pathlib.py:1183: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = PosixPath('/root/arrow/testing/data/flight/root-ca.pem')
name = '/root/arrow/testing/data/flight/root-ca.pem', flags = 524288, mode = 438

    def _opener(self, name, flags, mode=0o666):
        # A stub for the opener argument to built-in open()
>       return self._accessor.open(self, flags, mode)

../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/pathlib.py:1037: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

pathobj = PosixPath('/root/arrow/testing/data/flight/root-ca.pem'), args = (524288, 438)

    @functools.wraps(strfunc)
    def wrapped(pathobj, *args):
>       return strfunc(str(pathobj), *args)
E       FileNotFoundError: [Errno 2] No such file or directory: '/root/arrow/testing/data/flight/root-ca.pem'

../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/pathlib.py:387: FileNotFoundError

During handling of the above exception, another exception occurred:

    @pytest.mark.requires_testing_data
    def test_mtls():
        """Test mutual TLS (mTLS) with gRPC."""
>       certs = example_tls_certs()

../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/site-packages/pyarrow/tests/test_flight.py:1724: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/site-packages/pyarrow/tests/test_flight.py:89: in example_tls_certs
    "root_cert": read_flight_resource("root-ca.pem"),
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

path = 'root-ca.pem'

    def read_flight_resource(path):
        """Get the contents of a test resource file."""
        root = resource_root()
        if not root:
            return None
        try:
            with (root / path).open("rb") as f:
                return f.read()
        except FileNotFoundError:
            raise RuntimeError(
                "Test resource {} not found; did you initialize the "
                "test resource submodule?\n{}".format(root / path,
>                                                     traceback.format_exc()))
E           RuntimeError: Test resource /root/arrow/testing/data/flight/root-ca.pem not found; did you initialize the test resource submodule?
E           Traceback (most recent call last):
E             File "/tmp/arrow-6.0.1.edIR4/test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/site-packages/pyarrow/tests/test_flight.py", line 77, in read_flight_resource
E               with (root / path).open("rb") as f:
E             File "/tmp/arrow-6.0.1.edIR4/test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/pathlib.py", line 1183, in open
E               opener=self._opener)
E             File "/tmp/arrow-6.0.1.edIR4/test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/pathlib.py", line 1037, in _opener
E               return self._accessor.open(self, flags, mode)
E             File "/tmp/arrow-6.0.1.edIR4/test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/pathlib.py", line 387, in wrapped
E               return strfunc(str(pathobj), *args)
E           FileNotFoundError: [Errno 2] No such file or directory: '/root/arrow/testing/data/flight/root-ca.pem'

../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/site-packages/pyarrow/tests/test_flight.py:83: RuntimeError
__________________________________________ test_generic_options __________________________________________

path = 'root-ca.pem'

    def read_flight_resource(path):
        """Get the contents of a test resource file."""
        root = resource_root()
        if not root:
            return None
        try:
>           with (root / path).open("rb") as f:

../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/site-packages/pyarrow/tests/test_flight.py:77: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = PosixPath('/root/arrow/testing/data/flight/root-ca.pem'), mode = 'rb', buffering = -1
encoding = None, errors = None, newline = None

    def open(self, mode='r', buffering=-1, encoding=None,
             errors=None, newline=None):
        """
        Open the file pointed by this path and return a file object, as
        the built-in open() function does.
        """
        if self._closed:
            self._raise_closed()
        return io.open(str(self), mode, buffering, encoding, errors, newline,
>                      opener=self._opener)

../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/pathlib.py:1183: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = PosixPath('/root/arrow/testing/data/flight/root-ca.pem')
name = '/root/arrow/testing/data/flight/root-ca.pem', flags = 524288, mode = 438

    def _opener(self, name, flags, mode=0o666):
        # A stub for the opener argument to built-in open()
>       return self._accessor.open(self, flags, mode)

../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/pathlib.py:1037: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

pathobj = PosixPath('/root/arrow/testing/data/flight/root-ca.pem'), args = (524288, 438)

    @functools.wraps(strfunc)
    def wrapped(pathobj, *args):
>       return strfunc(str(pathobj), *args)
E       FileNotFoundError: [Errno 2] No such file or directory: '/root/arrow/testing/data/flight/root-ca.pem'

../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/pathlib.py:387: FileNotFoundError

During handling of the above exception, another exception occurred:

    @pytest.mark.requires_testing_data
    def test_generic_options():
        """Test setting generic client options."""
>       certs = example_tls_certs()

../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/site-packages/pyarrow/tests/test_flight.py:1882: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/site-packages/pyarrow/tests/test_flight.py:89: in example_tls_certs
    "root_cert": read_flight_resource("root-ca.pem"),
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

path = 'root-ca.pem'

    def read_flight_resource(path):
        """Get the contents of a test resource file."""
        root = resource_root()
        if not root:
            return None
        try:
            with (root / path).open("rb") as f:
                return f.read()
        except FileNotFoundError:
            raise RuntimeError(
                "Test resource {} not found; did you initialize the "
                "test resource submodule?\n{}".format(root / path,
>                                                     traceback.format_exc()))
E           RuntimeError: Test resource /root/arrow/testing/data/flight/root-ca.pem not found; did you initialize the test resource submodule?
E           Traceback (most recent call last):
E             File "/tmp/arrow-6.0.1.edIR4/test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/site-packages/pyarrow/tests/test_flight.py", line 77, in read_flight_resource
E               with (root / path).open("rb") as f:
E             File "/tmp/arrow-6.0.1.edIR4/test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/pathlib.py", line 1183, in open
E               opener=self._opener)
E             File "/tmp/arrow-6.0.1.edIR4/test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/pathlib.py", line 1037, in _opener
E               return self._accessor.open(self, flags, mode)
E             File "/tmp/arrow-6.0.1.edIR4/test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/pathlib.py", line 387, in wrapped
E               return strfunc(str(pathobj), *args)
E           FileNotFoundError: [Errno 2] No such file or directory: '/root/arrow/testing/data/flight/root-ca.pem'

../test-miniconda/envs/_verify_wheel-3.6m/lib/python3.6/site-packages/pyarrow/tests/test_flight.py:83: RuntimeError
============================================ warnings summary ============================================
```

Reply:
Thanks.
It's a bug of `verify-release-candidate.sh`. We should use the target source (6.0.1) directory instead of the current source (master) here: https://github.com/apache/arrow/blob/master/dev/release/verify-release-candidate.sh#L625

```shell
INSTALL_PYARROW=OFF ${ARROW_DIR}/ci/scripts/python_wheel_unix_test.sh ${source_dir not ARROW_DIR}
```

We should fix this as a separated task first and then update this pull request.

Reply:
Ok, created https://github.com/apache/arrow/pull/11735